### PR TITLE
switch from jupyter-telemetry to jupyter-events

### LINKED ISSUE #4803: Migrate from jupyter_telemetry to jupyter_events?
The `jupyter_telemetry` package is [said to be unmaintained](https://github.com/jupyter/telemetry/pull/69) and it test suite does not work with Python 3.12. It does not seem to be used heavily in JupyterHub, can you migrate to [jupyter_events](https://github.com/jupyter/jupyter_events) instead?

### PR DESCRIPTION
**Pull request by minrk**
Changes:

- id must be a URL
- config section is `EventLogger` instead of `EventLog`
- change `record_event` to `emit`

otherwise the same, really. Technically breaking since it affects config and emitted event ids

closes #4803

### CODE REVIEW

File: docs/source/reference/event-logging.md [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
 
-Event logging is handled by its `Eventlog` object. This leverages Python's standing [logging] library to emit, filter, and collect event data.
+Event logging is handled by its `EventLogger` object. This leverages Python's standing [logging] library to emit, filter, and collect event data.
 
 To begin recording events, you'll need to set two configurations:
 
-> 1. `handlers`: tells the EventLog _where_ to route your events. This trait is a list of Python logging handlers that route events to the event log file.
-> 2. `allows_schemas`: tells the EventLog _which_ events should be recorded. No events are emitted by default; all recorded events must be listed here.
+> 1. `handlers`: tells the EventLogger _where_ to route your events. This trait is a list of Python logging handlers that route events to the event log file.
+> 2. `allows_schemas`: tells the EventLogger _which_ events should be recorded. No events are emitted by default; all recorded events must be listed here.
```
Comment (manics) [inline docs/source/reference/event-logging.md:12]: Should `allowed_schemas` be deleted from this doc? https://github.com/jupyter/jupyter_events/pull/4 says
> "allowed_schemas", "allowed_categories", and "allowed_properties", have been removed from the EventLogger.

File: docs/source/conf.py [hunk 2]
```diff
@@ -289,6 +289,7 @@ def setup(app):
     r"https://github.com/[^/]*$",  # too many github usernames / searches in changelog
     "https://github.com/jupyterhub/jupyterhub/pull/",  # too many PRs in changelog
     "https://github.com/jupyterhub/jupyterhub/compare/",  # too many comparisons in changelog
+    "https://schema.jupyter.org/jupyterhub/.*",  # schemas are not published
```
Comment (manics) [inline docs/source/conf.py:292]: (Off topic: _should_ they be published?)

File: jupyterhub/tests/test_eventlog.py [hunk 3]
```diff
@@ -22,7 +22,7 @@
 # { ( '<schema id>', <version> ) : { <event_data> } }
 valid_events = [
     (
-        'hub.jupyter.org/server-action',
+        'https://schema.jupyter.org/jupyterhub/events/server-action',
         1,
```
Comment (manics) [inline jupyterhub/tests/test_eventlog.py:26]: The version field is no longer used

File: docs/source/conf.py (hunk 2 above)
Comment (minrk) [reply docs/source/conf.py:292]: Yes, probably. But the schemas repo isn't quite set up yet

File: jupyterhub/tests/test_eventlog.py (hunk 3 above)
Comment (minrk) [reply jupyterhub/tests/test_eventlog.py]: good catch, removed

File: docs/source/conf.py (hunk 2 above)
Comment (minrk) [reply docs/source/conf.py]: added "yet"

**Pull request merged**